### PR TITLE
Refactor task objects

### PR DESCRIPTION
**Pull request by juanmanuelramallo**
- Added a base object
- Updated existing task objects to inherit from the base object

### CODE REVIEW

File: lib/rsgem/tasks/base.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    #
+    # Base class for task objects.
+    # Child classes must implement the instance method +perform+.
+    #
+    class Base
+      attr_reader :context, :extras
+
+      def initialize(context:, **extras)
+        @context = context
+        @extras = extras
```
Comment (fedeagripa) [inline lib/rsgem/tasks/base.rb]: can you explain a bit about why you decided to name it `extras` , it sounds a bit awkward to me this name at first sight

**Review by jpascual1994 — APPROVED** [review]

File: lib/rsgem/tasks/base.rb (hunk 1 above)
Comment (juanmanuelramallo) [reply lib/rsgem/tasks/base.rb]: Any suggestions? It's just a set of optional arguments

Comment (fedeagripa) [reply lib/rsgem/tasks/base.rb]: args ?

Comment (juanmanuelramallo) [reply lib/rsgem/tasks/base.rb]: Done

**Review by fedeagripa — APPROVED** [review]

**Pull request merged**